client/src/document/ingredients/browser-compatibility-table/browser-info.tsx

```diff
@@ -0,0 +1,14 @@
+import React, { useContext } from "react";
+import type bcd from "mdn-browser-compat-data/types";
+
+export const BrowserInfoContext = React.createContext<bcd.Browsers | null>(
+  null
+);
+
+export function BrowserName({ id }: { id: bcd.BrowserNames }) {
```

Comment:
Why doesn't TS complain about a lack of return type?

Reply:
This is connected to your comment above. TS can infer the return type of functions. Which is why we don't need to generally declare them for React component functions. Unfortunately in this case we're not returning an element, so React would be confused about the return type. It's a TypeScript limitation: https://github.com/microsoft/TypeScript/issues/21699

Arguably one could go with [type coercion](https://www.typescriptlang.org/v2/en/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4BYAKCswFcA7DYCeuAZRimHoHMBRADZIQSejAAUASjgBvKnAVwiMWlFb4Azp2498cFBv30AnvsPJ0MAHQWMg4aJhUAvlSoAeDl172RYgPQAfEA) instead which has the upside of not adding runtime cost (small, but still) for a type system limitation

Reply:
Is this why you're doing `return <>{someString}</>` instead of `return someString`?

Reply:
Exactly. Instead of coercing the return type into one that TypeScript knows to expect in a JSX context (namely `ReactElement`), I just wrap it in a known (low-overhead) react element, which is Fragment.